code/datums/quirks/negative_quirks/csl.dm

```diff
@@ -0,0 +1,98 @@
+/datum/quirk/csl
+	name = "Common Second Language"
+	desc = "Common is not your native tongue - it's something you had to pick up along the way. \
+		Parsing common will be difficult (but not impossible), and you may drift back to your native tongue \
+		when you are stressed, anxious, or angry."
+	icon = FA_ICON_LANDMARK_DOME
+	value = -2
+	gain_text = span_danger("You have difficulty parsing Common.")
+	lose_text = span_notice("Common starts to click for you.")
+
+/datum/quirk/csl/add(client/client_source)
+	if(iscarbon(quirk_holder))
+		quirk_holder.remove_language(/datum/language/common, UNDERSTOOD_LANGUAGE, LANGUAGE_SPECIES)
+	else
+		quirk_holder.remove_language(/datum/language/common, UNDERSTOOD_LANGUAGE, LANGUAGE_ATOM)
+	quirk_holder.grant_partial_language(/datum/language/common, text2num(client_source?.prefs?.read_preference(/datum/preference/choiced/csl_strength)) || 90, type)
+	RegisterSignal(quirk_holder, COMSIG_SPECIES_GAIN, PROC_REF(reremove_common))
+	RegisterSignal(quirk_holder, COMSIG_MOB_SAY, PROC_REF(translate_everything))
+	RegisterSignal(quirk_holder, COMSIG_MOVABLE_LANGUAGE_BEING_TRANSLATED, PROC_REF(translate_parts))
+
+/datum/quirk/csl/remove()
+	UnregisterSignal(quirk_holder, COMSIG_SPECIES_GAIN)
+	UnregisterSignal(quirk_holder, COMSIG_MOB_SAY)
+	UnregisterSignal(quirk_holder, COMSIG_MOVABLE_LANGUAGE_BEING_TRANSLATED)
+
+	if(QDELING(quirk_holder))
+		return
+
+	quirk_holder.remove_partial_language(/datum/language/common, type)
+	var/mob/living/carbon/carbon_quirk_holder = quirk_holder
+	if(istype(carbon_quirk_holder) && carbon_quirk_holder.dna.species)
+		// only give back common if they're a species that should speak it
+		var/datum/language_holder/species_holder = GLOB.prototype_language_holders[carbon_quirk_holder.dna.species.species_language_holder]
+		if(LAZYACCESS(species_holder.spoken_languages, /datum/language/common))
+			quirk_holder.grant_language(/datum/language/common, UNDERSTOOD_LANGUAGE, LANGUAGE_SPECIES)
+	else
+		quirk_holder.grant_language(/datum/language/common, UNDERSTOOD_LANGUAGE, LANGUAGE_ATOM)
+
+/datum/quirk/csl/is_species_appropriate(datum/species/mob_species)
+	var/datum/language_holder/species_holder = GLOB.prototype_language_holders[mob_species.species_language_holder]
+	if(isnull(species_holder))
+		return FALSE
+	if(length(species_holder.spoken_languages) < 2)
+		return FALSE
+	return ..()
+
+/// Gets our native language from our list of spoken languages
+/datum/quirk/csl/proc/get_native_language()
+	var/list/language_pool = quirk_holder.get_language_holder()?.spoken_languages?.Copy()
+	if(!length(language_pool))
+		return // no languages to pick from at all?
+
+	// Don't want this
+	language_pool -= /datum/language/common
+	// If we have native languages set, prefer them
+	var/obj/item/organ/tongue/tongue = quirk_holder.get_organ_by_type(/obj/item/organ/tongue)
```

Comment:
I'm wondering if this might be a legitimate case for putting something on the species datum rather than the organ, it would be weird to me if changing your tongue changes the language you occasionally talk in especially as (if I understand correctly) it might change it to one you don't actually speak or understand?  
Or at the very least, maybe we should just cache this per quirk-owner based on what tongue they have at roundstart rather than every time we check.

Reply:
> it would be weird to me if changing your tongue changes the language you occasionally talk in especially as (if I understand correctly) it might change it to one you don't actually speak or understand?  

It can't pick a language you don't speak (see `language_pool &= languages_native` below). It just prioritizes languages your tongue can speak 

> Or at the very least, maybe we should just cache this per quirk-owner based on what tongue they have at roundstart rather than every time we check.

Maybe wise